OpenRA.Mods.Common/Widgets/TooltipContainerWidget.cs

```diff
@@ -25,23 +25,46 @@ public class TooltipContainerWidget : Widget
 
 		public Action BeforeRender = Nothing;
 		public int TooltipDelayMilliseconds = 200;
+		readonly Widget container;
```

Comment:
Do we need to add a container? TBH this change to me this additional widget seems completely unnecessary

Reply:
I don't think we add any other children to `TooltipContainerWidget`, and if we do then we just shouldn't

Reply:
We might as well call `RemoveChildren` on widget creation

Reply:
`IsVisible` is called from `Widget.Draw` as well as in `Draw` of `this`.

The container is needed to avoid a `EnumerationException` indicating the collection that was being iterated in `Widget.Draw` was being modified while enumerating. Using the child container solves this. At the cost of an extra widget level.

Reply:
`RemoveChildren` is called in `Tick` methods, not in `Draw` so there shouldn't be any issue with enumeration

Reply:
I don't think it's possible to get `EnumerationException` here. You are adding a child to the current widget, not the parent of the current widget

Reply:
You are right. An EnumerationException was triggered in the past - which caused me to add the container. It originated from `Widget.DrawOuter` where `IsVisible()` is called.  Perhaps i mistakenly added the tooltip to `Parent` rather than `this`. 

Also removed the call to `IsVisible` in `Draw` as `Draw` is only called if visible.